subsys/bluetooth/shell/bt.c

```diff
@@ -1147,14 +1152,85 @@ static int cmd_scan_filter_clear_addr(const struct shell *sh, size_t argc,
 #endif /* CONFIG_BT_OBSERVER */
 
 #if defined(CONFIG_BT_BROADCASTER)
-static const struct bt_data ad_discov[] = {
-	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
-};
+static void ad_mode_parse(const struct shell *sh, size_t argc, char *argv[], bool *discov)
+{
+	for (size_t argn = 0; argn < argc; argn++) {
+		const char *arg = argv[argn];
+
+		if (!strcmp(arg, "discov")) {
+			*discov = true;
+			break;
+		} else if (!strcmp(arg, "non_discov")) {
+			*discov = false;
+			break;
+		}
+	}
+}
+
+static ssize_t ad_init(struct bt_data *data, size_t data_size, bool discoverable)
+{
+	static uint8_t ad_flags;
+	size_t len = 0;
+	int err;
+
+	/* Set BR/EDR Not Supported if LE-only device */
+	ad_flags = IS_ENABLED(CONFIG_BT_BREDR) ? 0 : BT_LE_AD_NO_BREDR;
+
+	if (discoverable) {
+		ad_flags |= BT_LE_AD_GENERAL;
+	}
+
+	if (ad_flags != 0) {
+		__ASSERT(data_size > len, "No space for AD_FLAGS");
+		data[len].type = BT_DATA_FLAGS;
+		data[len].data_len = sizeof(ad_flags);
+		data[len].data = &ad_flags;
+		len++;
+	}
+
+#if defined(CONFIG_BT_AUDIO)
```

Comment:
Any reason for the `CONFIG_BT_AUDIO` guard here?

Reply:
it's because e.g. `audio.h` cannot be included because some of the struct members are dependent on Kconfig option that is available only if `CONFIG_BT_AUDIO` is enabled. E.g `bt_codec->data[CONFIG_BT_CODEC_MAX_DATA_COUNT]` Thus I had to guard the includes and some parts of the code.

Reply:
> it's because e.g. audio.h cannot be included because some of the struct members are dependent on Kconfig option that is available only if CONFIG_BT_AUDIO is enabled.

That should be fixed then :) Public header files should always be possible to include regardless of any Kconfig option.

 It's fine to postpone that fix to a later PR and remove this guard, but please add a comment/todo here then.

Reply:
I'll just add the fix as separate patch in this patchset